.github/workflows/publish.yaml

```diff
@@ -0,0 +1,72 @@
+name: Release
+on:
+  workflow_run:
+    workflows: ["Test"]
+    branches: [main]
+    types:
+      - completed
+jobs:
+  push_tag:
+    name: Push Tag
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-python@v3
+      - name: Push tag
+        id: tag_version
+        uses: mathieudutour/github-tag-action@v6.1
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+
+  build_wheels:
+    needs: push_tag
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, windows-2022, macOS-11]
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v3
+      - name: Install prerequisites
+        run: python -m pip install cibuildwheel==2.11.2
+
+      - name: Build wheels
+        run: |
+          git fetch --depth=1 origin +refs/tags/*:refs/tags/*
+          python -m cibuildwheel --output-dir wheelhouse
+
+      - uses: actions/upload-artifact@v3
+        with:
+          path: ./wheelhouse/*.whl
+
+  publish:
```

Comment:
https://github.com/pypa/gh-action-pypi-publish may be more straightforward

Reply:
I just used what you had from your branch since it was more "built for purpose." My file was copied from a personal project that had more concerns like compiling C++ code and ensuring separate wheels for different systems.